### PR TITLE
Localisation support

### LINKED ISSUE #12: Internationalization support
Unblocks: #8

### PR DESCRIPTION
**Pull request by mtaylor76**
Adds support for dynamic localisation to various languages using [i18next](https://www.i18next.com/) and [react-i18next,](https://react.i18next.com/) with [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector) to automatically detect the user's native OS language preferences and switch the app's language accordingly.

This PR also adds a Spanish translation, partly to demonstrate functionality and partly since the app need one anyways. Also, English text was cleaned up a bit as well.

Closes #12, unblocks #8 

## Adding new text to the localisations (for developers)

As features are added to the calculator, they'll most likely not be localised immediately. To localise any new text in the app, make sure to first wrap the text in the `i18n.t()` method. Then, in your language's `translation.json`, add a new JSON key/value pair with the key used in the `t()` method, using your translated text as the value.

For example, let's say we're adding a new button that says "Show Current Pattern". In the component where that text would be displayed, you would instead write `i18n.t("Show Current Pattern")`, wrapping the statement in brackets `{}` if it were part of JSX code. 

Then, to add a translation to your language, you'd go to your languages `translation.json` file, and add the following line (using Spanish as an example):

`"Show Current Pattern": "Mostrar Patrón Actual",`

If no key is specified in the localisation files, the app will simply display whatever text is passed into `i18n.t()`, to avoid errors.

## Creating new localisations

To localise the app to a new language, the following simple steps need to be done:
* Create a new folder under `/locales` for your language using the standard two-letter localisation code for your language (e.g. `es` for Spanish, `zh` for Chinese).
* In the newly created folder, copy and paste the `translation.json` file from one of the other languages.
* Edit the `translation.json` file to translate each JSON value to the appropriate translated value.
* In `/src/i18n.js`, create an `import` statement for your language, importing the `translation.json` file you just created. Follow the pattern used by the existing statements.
* In the same file (`/src/i18n.js`), add your language to the list of languages inside the `resources` object within the `i18n.init()` method, following the pattern of other declared languages.

The localisation is now ready to deploy and will appear to users who use your language for their browser.

## Demo (Spanish localisation)

image

### CODE REVIEW

**Comment by mtaylor76** [comment]
Files for French and German localisations have been added; all it will take to localise to those languages is to actually translate the text in their respective `translations.json` files.

**Comment by elxris** [comment]
Thank you so much for this.

**Comment by elxris** [comment]
Can you copy and paste your instructions in the Readme?

**Comment by mtaylor76** [comment]
> 
> 
> Can you copy and paste your instructions in the Readme?

Done!

File: src/Chart.jsx [hunk 1]
```diff
@@ -77,9 +78,9 @@ const ChartComponent = ({ filter }) => {
       type: "line",
       data: {
         datasets: generateData(filter),
-        labels: "Mon Tue Wed Thu Fri Sat"
+        labels: i18n.t("Mon Tue Wed Thu Fri Sat")
           .split(" ")
-          .reduce((acc, day) => [...acc, `${day} AM`, `${day} PM`], []),
+          .reduce((acc, day) => [...acc, `${day} `+i18n.t('AM'), `${day} `+i18n.t('PM')], []),
```
Comment (elxris) [inline src/Chart.jsx:83]:
```suggestion
          .reduce((acc, day) => [...acc, `${day}  ${i18n.t('AM')}`, `${day} ${i18n.t('PM')}`], []),
```

File: src/Filter.jsx [hunk 2]
```diff
@@ -32,16 +33,16 @@ const ClearButton = (props) => {
...
 };
 
 const names = [
-  "Buy Price",
-  ..."Mon Tue Wed Thu Fri Sat"
+  i18n.t("Buy Price"),
+  ...i18n.t("Mon Tue Wed Thu Fri Sat")
     .split(" ")
-    .reduce((curr, day) => [...curr, ...[`${day} AM`, `${day} PM`]], []),
+    .reduce((curr, day) => [...curr, ...[`${day} `+i18n.t('AM'), `${day} `+i18n.t('PM')]], []),
```
Comment (elxris) [inline src/Filter.jsx:45]:
```suggestion
    .reduce((curr, day) => [...curr, ...[`${day} ${i18n.t('AM')}`, `${day} ${i18n.t('PM')}`]], []),
```

**Comment by elxris** [comment]
Thank you for all!
This will come out today!

**Pull request merged**